plugins/generator-1.15.2/forge-1.15.2/templates/plant.java.ftl

```diff
@@ -413,29 +412,35 @@ import net.minecraft.block.material.Material;
 					int z = pos.getZ();
 					additionalCondition = <@procedureOBJToConditionCode data.placingCondition/>;
 				}
+
+				Block block = state.getBlock();
+
 				</#if>
 				return
 				<#if (data.canBePlacedOn?size > 0)>(
-				<#list data.canBePlacedOn as canBePlacedOn>
-					block == ${mappedBlockToBlockStateCode(canBePlacedOn)}.getBlock()
-					<#if canBePlacedOn?has_next>||</#if>
-				</#list>)</#if>
+					<#list data.canBePlacedOn as canBePlacedOn>
+						block == ${mappedBlockToBlockStateCode(canBePlacedOn)}.getBlock()
+						<#if canBePlacedOn?has_next>||</#if>
+					</#list>)
+				</#if>
 				<#if (data.canBePlacedOn?size > 0) && hasCondition(data.placingCondition)> && </#if>
 				<#if hasCondition(data.placingCondition)> additionalCondition </#if>;
 			}
 			</#if>
 
 			@Override public boolean isValidPosition(BlockState state, IWorldReader worldIn, BlockPos pos) {
-				BlockPos blockpos = pos.down();
-				BlockState blockstate = worldIn.getBlockState(blockpos);
-				Block block = blockstate.getBlock();
 				<#if hasCondition(data.placingCondition) && data.plantType == "growapable">
 				World world = worldIn.getDimension().getWorld();
 				int x = pos.getX();
 				int y = pos.getY();
 				int z = pos.getZ();
 				boolean additionalCondition = <@procedureOBJToConditionCode data.placingCondition/>;
 				</#if>
+
+				BlockPos blockpos = pos.down();
+				BlockState blockstate = worldIn.getBlockState(blockpos);

```

Comment:
You can't place the blockpos declaration after because before it, coordinates use it to be declared.

Reply:
Coordinates in condition if use BlockPos, or which part?

Reply:
My bad, they use `pos` which is already declared, it's `blockpos` that isn't declared.